src/Features/Core/Portable/Common/DocumentNavigationOperation.cs

```diff
@@ -21,25 +19,17 @@ namespace Microsoft.CodeAnalysis.CodeActions
     /// </summary>
     public class DocumentNavigationOperation : CodeActionOperation
     {
-        private readonly DocumentId _documentId;
-        private readonly int _position;
+        internal DocumentId DocumentId { get; }
+        internal int Position { get; }
 
-        public DocumentNavigationOperation(DocumentId documentId, int position = 0)
+        public DocumentNavigationOperation(DocumentId documentId!!, int position = 0)
         {
-            _documentId = documentId ?? throw new ArgumentNullException(nameof(documentId));
-            _position = position;
+            DocumentId = documentId;
+            Position = position;
         }
 
         public override void Apply(Workspace workspace, CancellationToken cancellationToken)
         {
-            if (workspace.CanOpenDocuments)
-            {
-                var navigationService = workspace.Services.GetService<IDocumentNavigationService>();
-                var threadingService = workspace.Services.GetService<IWorkspaceThreadingServiceProvider>();
```

Comment:
Can you delete IWorkspaceThreadingServiceProvider or did something else still use it too?

Reply:
unfortunately, unit testing and TS are still using it in exactly 2 locations.